File: packages/node-definitions/src/trigger.ts
import type { NodeDefinition } from './types'

export const WebhookTriggerNode: NodeDefinition = {
  type: 'trigger.webhook',
  name: 'Webhook Trigger',
  category: 'trigger',
  description: 'Trigger workflow via HTTP POST request',
  icon: 'Zap',
  color: '#10b981',
  properties: [
    {
      name: 'path',
      label: 'Webhook Path',
      type: 'string',
      default: 'webhook-id',
      required: true,
    },
  ],
  inputs: 0,
  outputs: 1,
}

File: apps/api/app/node_system/builtins/trigger.py
from typing import Any

from apps.api.app.node_system.base.base_node import BaseNode
from apps.api.app.node_system.base.node_context import NodeContext
from apps.api.app.node_system.base.node_metadata import NodeMetadata
from apps.api.app.node_system.base.node_result import NodeResult

class TriggerNode(BaseNode):
    @classmethod
    def get_metadata(cls) -> NodeMetadata:
        return NodeMetadata(
            type="trigger.manual",
            name="Start",
            category="trigger",
            description="Initiate workflow execution manually",
            properties=[
                {
                    "name": "startWorkflow",
                    "label": "Start Workflow",
                    "type": "string",
                    "default": "manual",
                }
            ],
            inputs=0,
            outputs=1,
        )

    async def execute(self, input_data: dict[str, Any], context: NodeContext) -> NodeResult:
        # Starter nodes receive the initial workflow input (from context or input_data)
        return NodeResult(
            success=True,
            output_data=input_data,
        )
